nil pointer dereference after about 10 sec

Hi!
i was trying out the new enrich future when i suddenly got an nil pointer dereference. This is about 10 sec after i have started skogul, and  in this instance there is nothing streaming in on port 50k


```
sveinov@stc:~/skogul$ ./skogul -f /etc/skogul/default.json 
WARN[0000] deprecation warning for transformer "extract_if_name": ExtractFromData is replaced by CopyFromData. ExtractFromData will be removed in future versions.  category=core core=config
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9b34e4]

goroutine 40 [running]:
github.com/telenornms/skogul/parser.(*ProtoBuf).GetStats(0xc00036e690)
	/home/sveinov/skogul/parser/protobuf.go:246 +0x2a4
github.com/telenornms/skogul/stats.Collect(...)
	/home/sveinov/skogul/stats/chan.go:95
main.startStats(0xc000347980)
	/home/sveinov/skogul/cmd/skogul/main.go:203 +0x2c6
created by main.main
	/home/sveinov/skogul/cmd/skogul/main.go:184 +0x692


```

Here is the error where i have some incoming data:

```

sveinov@stc:~/skogul$ ./skogul -f /etc/skogul/default.json 
WARN[0000] deprecation warning for transformer "extract_if_name": ExtractFromData is replaced by CopyFromData. ExtractFromData will be removed in future versions.  category=core core=config
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0xaf834b]

goroutine 15 [running]:
github.com/telenornms/skogul/sender.checkVariable({0xdc5f06, 0x8}, {0xdc4a4f, 0x5}, {0xc0002a66c0, 0x6}, {0x0?, 0x0?})
	/home/sveinov/skogul/sender/influxdb.go:64 +0x8b
github.com/telenornms/skogul/sender.(*InfluxDB).Send(0xc0002d1e30, 0xc000e22060)
	/home/sveinov/skogul/sender/influxdb.go:139 +0xd7b
github.com/telenornms/skogul.(*Handler).Send(0xc0002bd460, 0xc000e22060?)
	/home/sveinov/skogul/common.go:299 +0x8b
github.com/telenornms/skogul.(*Handler).TransformAndSend(0xc0002bd460?, 0xc000834000?)
	/home/sveinov/skogul/common.go:324 +0x88
github.com/telenornms/skogul.(*Handler).Handle(0x0?, {0xc000834000?, 0x0?, 0x0?})
	/home/sveinov/skogul/common.go:312 +0x3d
github.com/telenornms/skogul/receiver.(*UDP).process(0xc000050c80)
	/home/sveinov/skogul/receiver/udp.go:81 +0xc5
created by github.com/telenornms/skogul/receiver.(*UDP).Start
	/home/sveinov/skogul/receiver/udp.go:123 +0x19c
```



It is fresh from the main branch, but i have tried different stable versions and get the same result.
``
commit b97fe08f6218db41862bafa35707a3dc1bd43578 (HEAD -> primary, origin/primary, origin/HEAD)
Skogul 0.16.03gb97fe08dirty built using go1.18.2
``


Config:
[skogul_conf.txt](https://github.com/telenornms/skogul/files/8823888/skogul_conf.txt)

--- a/parser/protobuf.go
+++ b/parser/protobuf.go
@@ -57,21 +57,22 @@ type protobufStats struct {
 	Parsed                       uint64 // Successful parses
 }
 
+func (x *ProtoBuf) initStats() {
+	x.stats = &protobufStats{
+		Received:                     0,
+		ParseErrors:                  0,
+		MissingExtension:             0,
+		FailedToCastToJuniperMessage: 0,
+		FailedToJsonMarshal:          0,
+		FailedToJsonUnmarshal:        0,
+		NilData:                      0,
+		Parsed:                       0,
+	}
+}
+
 // Parse accepts a byte slice of protobuf data and marshals it into a container
 func (x *ProtoBuf) Parse(b []byte) (*skogul.Container, error) {
-	x.once.Do(func() {
-		// XXX: This doesn't start until the first message is parsed. But it's probably fine for now.
-		x.stats = &protobufStats{
-			Received:                     0,
-			ParseErrors:                  0,
-			MissingExtension:             0,
-			FailedToCastToJuniperMessage: 0,
-			FailedToJsonMarshal:          0,
-			FailedToJsonUnmarshal:        0,
-			NilData:                      0,
-			Parsed:                       0,
-		}
-	})
+	x.once.Do(x.initStats)
 	atomic.AddUint64(&x.stats.Received, 1)
 	parsedProtoBuf, err := parseTelemetryStream(b)
 
@@ -242,6 +243,9 @@ func (x *ProtoBuf) GetStats() *skogul.Metric {
 	metric.Metadata["component"] = "parser"
 	metric.Metadata["type"] = "protobuf"
 	metric.Metadata["identity"] = skogul.Identity[x]
+
+	// Ensure we init the stats struct in case we havent received a message yet.
+	x.once.Do(x.initStats)
 
 	metric.Data["received"] = x.stats.Received
 	metric.Data["parse_errors"] = x.stats.ParseErrors

